ta/pkcs11/src/processing_ec.c

```diff
@@ -646,3 +646,102 @@ size_t ecdsa_get_input_max_byte_size(TEE_OperationHandle op)
 		return 0;
 	}
 }
+
+enum pkcs11_rc pkcs2tee_param_ecdh(struct pkcs11_attribute_head *proc_params,
+				   void **pub_data, size_t *pub_size)
+{
+	enum pkcs11_rc rc = PKCS11_CKR_GENERAL_ERROR;
+	struct serialargs args = { };
+	uint32_t temp = 0;
+
+	serialargs_init(&args, proc_params->data, proc_params->size);
+
+	/* Skip KDF */
+	rc = serialargs_get(&args, &temp, sizeof(uint32_t));
+	if (rc)
+		return rc;
+
+	/* Shared data size, shall be 0 */
+	rc = serialargs_get(&args, &temp, sizeof(uint32_t));
+	if (rc || temp)
+		return rc;
+
+	/* Public data size and content */
+	rc = serialargs_get(&args, &temp, sizeof(uint32_t));
+	if (rc || !temp)
+		return rc;
+
+	*pub_size = temp;
+
+	rc = serialargs_get(&args, &temp, sizeof(uint8_t));
+	if (rc)
+		return rc;
+
+	if (temp != 0x02 && temp != 0x03 && temp != 0x04)
+		return PKCS11_CKR_ARGUMENTS_BAD;
+
+	if (temp != 04) {
```

Comment:
something fishy in here. `temp` is the same as what was above `if` and here we have octal 04 which is same as hex 0x04 above.

Please see `tee2pkcs_ec_attributes()` for an idea how DER encoded form has compression specified.

Also:
> A token MUST be able to accept this value encoded as a raw octet string (as per section A.5.2 of [ANSI X9.62]).  A token MAY, in addition, support accepting this value as a DER-encoded ECPoint (as per section E.6 of [ANSI X9.62]) i.e. the same as a CKA_EC_POINT encoding.  The calling application is responsible for converting the offered public key to the compressed or uncompressed forms of these encodings if the token does not support the offered form.

Reply:
right, the check should have been temp!= 0x04